File: haskell/015/Main.hs
{-
    ProjectEuler 015
    Haskell / Main.hs
    copyright (c) 2015 Susisu

    GHC 7.8.3
    `ghc Main.hs -o bin/main`
-}

module Main where

import Data.List

main :: IO ()
main = print $ combination 40 20

combination :: Int -> Int -> Int
combination n r = foldl' (\a (b, c) -> a * b `div` c) 1 $ zip (drop r [1 .. n]) [1 .. r]

File: javascript/015/main.js
/*
 * ProjectEuler 015
 * JavaScript / main.js
 * copyright (c) 2015 Susisu
 *
 * Node.js v4.2.1
 * `node main.js`
 */

"use strict";

function main() {
    console.log(combination(40, 20));
}

function combination(n, r) {
    var p = 1;
    for (var i = 1; i <= r; i++) {
        p *= n - r + i;
        p /= i;
    }
    return p;
}

main();
